mindmeld/rasaconverter.py

```diff
@@ -0,0 +1,545 @@
+            f.close()
+
+    @staticmethod
+    def __does_intent_ex_contain_entity(intent_example):
+        return len(re.findall(r"\[.*\]\(.*\)", intent_example)) > 0
+
+    @staticmethod
+    def __write_intent_with_extinty(intent_f, intent_example):
+        mindmend_intent_example = intent_example
+        for match in re.findall(r"\[.*\]\(.*\)", intent_example):
+            mindmeld_entity = match.replace("[", "{").replace("]", "|") \
+                .replace("(", "").replace(")", "}")
+            mindmend_intent_example = mindmend_intent_example.replace(match, mindmeld_entity)
+            intent_f.write(mindmend_intent_example)
+            # add this to the respective entity gazetteer file as well
+            Rasaconverter.create_entity_files(mindmeld_entity)
+
+    @staticmethod
+    def __remove_comments_from_line(line):
+        start_of_comment = line.find("<!---")
+        end_of_comment = line.find("-->")
+        line_without_comment = line.replace(line[start_of_comment:end_of_comment+3], '')
+        line_without_comment = line_without_comment.rstrip()
+        return line_without_comment
+
+    @staticmethod
+    def __add_example_to_training_file(current_intent_path, line):
+        with open(current_intent_path + "/train.txt", "a") as intent_f:
+            intent_example = line[2:]
+            intent_example = Rasaconverter.__remove_comments_from_line(intent_example) + "\n"
+            if Rasaconverter.__does_intent_ex_contain_entity(intent_example):
+                Rasaconverter.__write_intent_with_extinty(intent_f, intent_example)
+            else:
+                intent_f.write(intent_example)
+
+    def __read_domain_file(self):
+        if os.path.isfile(self.rasa_project_directory + "/domain.yml"):
+            try:
+                with open(self.rasa_project_directory + "/domain.yml", "r") as stream:
+                    domain_data_loaded = yaml.safe_load(stream)
+                    return domain_data_loaded
+            except IOError as e:
+                print("Can not open domain.yml file at %s"
+                      % self.rasa_project_directory + "/domain.yml")
+                print(e)
+        elif os.path.isfile(self.rasa_project_directory + "/domain.yaml"):
+            try:
+                with open(self.rasa_project_directory + "/domain.yml", "r") as stream:
+                    domain_data_loaded = yaml.safe_load(stream)
+                    return domain_data_loaded
+            except IOError as e:
+                print("Can not open domain.yml file at %s"
+                      % self.rasa_project_directory + "/domain.yaml")
+                print(e)
+        else:
+            print("Could not find domain.yml file in project directory")
+            raise FileNotFoundError
+
+    def __read_entities(self):
+        domain_file = self.__read_domain_file()
+        return domain_file['entities']
+
+    def __read_slots(self):
```

Comment:
```suggestion
    def _read_slots(self):
```

if you are indicating private method, a single _ is sufficient